go/app/infra.go

```
@@ -33,17 +44,146 @@ func NewItemRepository() ItemRepository {
 	return &itemRepository{fileName: "items.json"}
 }
 
+// JSONファイルの内容をパースするための構造体
+type ItemsData struct {
+	Items []Item `json:"items"`
+}
+
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
 	// STEP 4-1: add an implementation to store an item
 
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// 絶対パスを作成
+	filePath := cwd + "/" + i.fileName
+
+	// JSONファイルを開く (なければ作成)
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
```

If the JSON file doesn't exist, we should throw an error I think.